cmd/oracled/cmd/start.go

```diff
@@ -41,41 +37,20 @@ func startCmd() *cobra.Command {
 				return fmt.Errorf("failed to start event subscription: %w", err)
 			}
 
-			errChan := make(chan error, 1)
-			sigChan := make(chan os.Signal, 1)
-
-			srv := server.New(svc)
-
-			go func() {
-				if err := srv.Run(); err != nil {
-					if !errors.Is(err, http.ErrServerClosed) {
-						errChan <- err
-					} else {
-						close(errChan)
-					}
-				}
-			}()
+			errChan := server.Serve(svc)
 
+			sigChan := make(chan os.Signal, 1)
 			signal.Notify(sigChan, syscall.SIGTERM, syscall.SIGINT)
 
 			select {
 			case err := <-errChan:
 				if err != nil {
-					log.Errorf("http server was closed with an error: %v", err)
+					log.Errorf("rpc server was closed with an error: %v", err)
 				}
 			case <-sigChan:
 				log.Info("signal detected")
```

Comment:
it seems that you should close components here, which were created by `server.Serve(svc)`.

Reply:
OK. I added here to call close of grpc and gateway server.